python/pyspark/pandas/namespace.py

```diff
@@ -3399,8 +3399,8 @@ def resolve(internal: InternalFrame, side: str) -> InternalFrame:
         left_join_on_columns = [scol_for(left_table, label) for label in left_join_on_names]
         right_join_on_columns = [scol_for(right_table, label) for label in right_join_on_names]
         on = reduce(
-            lambda l, r: l & r,
-            [l == r for l, r in zip(left_join_on_columns, right_join_on_columns)],
+            lambda lft, r: lft & r,
+            [lft == r for lft, r in zip(left_join_on_columns, right_join_on_columns)],
```

Comment:
We could convert the other argument to something like `rgt`.

Current naming kind of suggests there is some asymmetry here and `r ` is different type of object from `lft`.